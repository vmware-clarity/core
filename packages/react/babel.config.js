/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */

// used by our jest test suite; not used for running the dev app
module.exports = {
  presets: [['@babel/preset-env', { targets: { esmodules: true } }]],
};
