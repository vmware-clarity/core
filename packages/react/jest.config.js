/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */

module.exports = {
  roots: ['./'],
  globals: {
    'ts-jest': {
      tsconfig: 'tsconfig.test.json',
    },
  },
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  moduleNameMapper: {
    '@cds/(.*)': '<rootDir>/../core/dist/$1',
    '../converter/react-wrapper.js': '<rootDir>/dist/react/converter/react-wrapper.js',
    '.*(utils/index.js)$': '<rootDir>/dist/react/utils/index.js',
  },
  setupFilesAfterEnv: ['./jest.setup.ts'],
  testEnvironment: 'jest-environment-jsdom',
  testMatch: ['**/__tests__/**/*.+(ts|tsx|js)', '**/?(*.)+(test).+(ts|tsx|js)'],
  transform: {
    '^.+\\.(js|jsx)$': 'babel-jest',
    '^.+\\.(ts|tsx)$': 'ts-jest',
  },
  transformIgnorePatterns: ['node_modules/(?!@cds/core|lit|@lit|@lit-labs|ramda)'],
};
