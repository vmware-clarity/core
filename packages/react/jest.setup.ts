/*
 * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */

import '@testing-library/jest-dom/extend-expect';

if (typeof window !== 'undefined') {
  // jsdom doesn't implment IntersectionObserver
  // https://github.com/jsdom/jsdom/issues/2032
  const mockObserverAPI = () =>
    jest.fn().mockReturnValue({
      observe: jest.fn(),
      unobserve: jest.fn(),
      disconnect: jest.fn(),
    });

  window.ResizeObserver = mockObserverAPI();
  window.IntersectionObserver = mockObserverAPI();

  // jsdom implements getBoundingClientRect, but not DOMRect
  if (!window.DOMRect) {
    const domRectMock = () => jest.fn().mockReturnValue({});
    window.DOMRect = domRectMock() as unknown as typeof window.DOMRect;
  }
}
