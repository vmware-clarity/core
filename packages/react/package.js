'use strict';

const fs = require('fs-extra');
const path = require('path');
const del = require('del');
const cpy = require('cpy');

const read = dir =>
  fs
    .readdirSync(dir)
    .reduce(
      (files, file) =>
        fs.statSync(path.join(dir, file)).isDirectory()
          ? files.concat(read(path.join(dir, file)))
          : files.concat(path.join(dir, file)),
      []
    );

function removeCacheFiles() {
  del.sync(
    [
      './dist/react/**/*.{tsbuildinfo,test.js,test.js.map,test.d.ts,.snap}',
      './dist/react/*.{tsbuildinfo,test.js,test.js.map,test.d.ts,.snap}',
    ],
    { force: true }
  );
}

function copyAssets() {
  return Promise.all([
    cpy(['./**/package.json'], './../dist/react/', { cwd: './src', parents: true }),
    cpy(['./package.json'], './dist/react/', { cwd: './', parents: true }),
    cpy(['./README.md'], './dist/react/', { cwd: './', parents: true }),
  ]);
}

function cleanPackageFiles() {
  read('./dist/react')
    .filter(f => f.includes('package.json'))
    .forEach(file => {
      const packageFile = fs.readJsonSync(file);
      ['alias', 'browserslist', 'scripts', 'devDependencies'].forEach(p => delete packageFile[p]);

      fs.writeJsonSync(file, { ...packageFile }, { spaces: 2 });
    });
}

(async () => {
  await copyAssets();
  removeCacheFiles();
  cleanPackageFiles();
})();
