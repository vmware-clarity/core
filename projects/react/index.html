<html>
  <body cds-text="body" cds-layout="p:lg">
    <div id="root"></div>
    <script src="./index.tsx"></script>
  </body>
</html>
