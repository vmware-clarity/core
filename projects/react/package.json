{
  "name": "react-ws",
  "private": true,
  "scripts": {
    "build": "npm-run-all _build:ts _build:package",
    "build:watch": "tsc --b ./tsconfig.project.json --watch",
    "clean": "rimraf ./dist",
    "start": "npm-run-all build --parallel build:watch _dev:start",
    "test": "jest --coverage",
    "test:watch": "jest --watchAll",
    "update-snapshots": "npm run build && jest --updateSnapshot",
    "_build:package": "cpy package.lib.json ./dist/lib --rename package.json && cpy README.md ./dist/lib && cpy \"./**/package.json\" ./../dist/lib --cwd=./src --parents && del-cli ./dist/**/*.tsbuildinfo",
    "_build:ts": "tsc --build ./tsconfig.project.json --force",
    "_dev:start": "parcel index.html"
  },
  "devDependencies": {
    "@babel/core": "7.10.2",
    "@babel/preset-env": "7.10.2",
    "@cds/core": "file:./../core/dist/lib",
    "@lit-labs/react": "^1.0.1",
    "@types/enzyme": "^3.10.5",
    "@types/enzyme-adapter-react-16": "^1.0.6",
    "@types/jest": "^26.0.0",
    "@types/node": "^14.0.5",
    "@types/react": "^16.9.35",
    "@types/react-dom": "^16.9.8",
    "babel-jest": "26.0.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.2",
    "enzyme-to-json": "^3.5.0",
    "jest": "^26.0.1",
    "lit": "2.0.2",
    "parcel": "^2.0.0-beta.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "replace": "1.1.0",
    "ts-jest": "^26.1.0",
    "typescript": "~4.2.4"
  }
}
