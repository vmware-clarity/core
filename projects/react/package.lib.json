{
  "name": "@cds/react",
  "description": "Core component modules for Clarity React",
  "homepage": "https://core.clarity.design",
  "keywords": ["react", "design system", "clarity", "ui"],
  "repository": {
    "type": "git",
    "url": "https://github.com/vmware-clarity/core.git"
  },
  "author": "clarity",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/vmware-clarity/core/issues"
  },
  "main": "./index.js",
  "module": "./index.js",
  "typings": "./index.d.ts",
  "type": "module",
  "dependencies": {
    "@lit-labs/react": "^1.0.1"
  },
  "peerDependencies": {
    "react": ">= 16.13.1",
    "react-dom": ">= 16.13.1",
    "@cds/core": "*"
  },
  "optionalDependencies": {
    "@cds/city": "^1.1.0",
    "modern-normalize": "^1.1.0"
  }
}
