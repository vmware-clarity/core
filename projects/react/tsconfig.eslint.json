{
  "compilerOptions": { "strict": true },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["dist", "node_modules"]
}
