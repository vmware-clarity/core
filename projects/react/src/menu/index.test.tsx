import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { CdsMenu, CdsMenuItem } from './index';

describe('CdsMenu', () => {
  it('renders', async () => {
    render(
      <>
        <CdsMenu>
          <CdsMenuItem>item 1</CdsMenuItem>
          <CdsMenuItem>item 2</CdsMenuItem>
        </CdsMenu>
      </>
    );

    expect((await screen.findAllByRole('menu')).length).toEqual(1);
    expect(document.querySelectorAll('cds-menu').length).toEqual(1);
    expect(document.querySelectorAll('cds-menu-item').length).toEqual(2);
  });

  it('snapshot', () => {
    const { container } = render(
      <>
        <CdsMenu>
          <CdsMenuItem>item 1</CdsMenuItem>
          <CdsMenuItem>item 2</CdsMenuItem>
        </CdsMenu>
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
