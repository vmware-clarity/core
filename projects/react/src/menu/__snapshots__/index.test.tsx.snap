// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdsMenu snapshot 1`] = `
<div>
  <cds-menu
    aria-orientation="vertical"
    role="menu"
  >
    <cds-menu-item
      tabindex="0"
    >
      item 1
    </cds-menu-item>
    <cds-menu-item
      tabindex="0"
    >
      item 2
    </cds-menu-item>
  </cds-menu>
</div>
`;
